pandas/core/arrays/datetimes.py

```diff
@@ -2146,7 +2146,9 @@ def objects_to_datetime64ns(
             require_iso8601=require_iso8601,
             allow_mixed=allow_mixed,
         )
-        result = result.reshape(data.shape, order=order)
+        # error: No overload variant of "reshape" of "_ArrayOrScalarCommon"
+        # matches argument types "Tuple[int, ...]", "str"
+        result = result.reshape(data.shape, order=order)  # type: ignore[call-overload]
```

Comment:
added ignore not needed. order is inferred as `str` by mypy. can override the mypy inference with a variable type annotation.

i.e. `order: Literal["F", "C"] = "F" if flags.f_contiguous else "C"`

Reply:
wont using Literal here cause problems?  IIUC the `__future__` import makes Literal OK i function annotations, but not variable annotations

Reply:
I _think_ variable annotations (inside functions) aren't parsed by the Python interpreter even without the \_\_future__ import, but will need to confirm this.

Reply:
yep. This works.

```
def func():
    a: Unknown
    print("Success!")

func()
```

Reply:
good call, updated